edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/PodExtensions.cs

```diff
@@ -43,31 +43,35 @@ static Option<V1ContainerStatus> GetContainerByName(string name, V1Pod pod)
             return Option.Maybe(status);
         }
 
-        static ReportedModuleStatus ConvertPodStatusToModuleStatus(Option<V1ContainerStatus> podStatus)
+        static ReportedModuleStatus ConvertPodStatusToModuleStatus(V1PodStatus podStatus)
         {
-            return podStatus.Map(
-                pod =>
+            ModuleStatus moduleStatus = ModuleStatus.Failed;
+            string description = "Invalid pod status";
+
+            if (podStatus?.Phase != null)
```

Comment:
Mike was talking about that we should reduce the usage of null in our codebase. Can you please check that we always have a podStatus and Phase in it. If yes, you can remove redundant check that only adds lines of code without bringing clarity; or change it to use `Option` type and deal with `None` explicitly.